### PR TITLE
Ticket3439 icon crediting

### PR DESCRIPTION
**Pull request by aaron-long**
### Description of work

After the E3 -> E4 migration 90% of icons used in the ibex_gui did not have a listed licence in the source code or the product licencing. I replaced all icons with new ones, attempting to retain the original themes but with a more consitent style. An 'Icon Licencing' dialog was created and is available from the help menu of the ibex gui. This licences file is located at ui.help.resources, and contains all licences for icons used in the the gui. I updated the developer wiki to also include a short section on this: https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/GUI-Other

### Ticket

https://waffle.io/ISISComputingGroup/IBEX/cards/5b6d68ad83dc89001e756e45

### Acceptance criteria

All icons used by the ibex gui have now been updated with licencing information. A new document is available to track all future new/changed licencing for icons.

### Unit tests

None

### System tests

None

### Documentation

I new page (Icon Licences) was included in the GUI Other section so developers will know how to properly credit authors:
https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/GUI-Other

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.ui.help/resources/iconlicences.txt [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+alarm designed by Freepik at www.flaticon.com
+refresh designed by Freepik at www.flaticon.com
+alarm_bell_24x24 designed by Freepik at www.flaticon.com
+alarm_bell_32x24 designed by Freepick at www.flaticon.com
+
+--
+
+Blocks
+
+->ibax.uk.blocks/icons
```
Comment (LiamPanchaud) [inline base/uk.ac.stfc.isis.ibex.ui.help/resources/iconlicences.txt]: ibax?

File: base/uk.ac.stfc.isis.ibex.ui.help/resources/iconlicences.txt [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+->ui.perspectiveswitcher/icons
+reset icon designed by Freepik at www.flaticon.com 
+
+
+
+Scripting
+
+->ui.scripting/icons
+alarm icon designed by Freepik from www.flaticon.com
```
Comment (LiamPanchaud) [inline base/uk.ac.stfc.isis.ibex.ui.help/resources/iconlicences.txt]: Maybe it's a good idea to explicitly give the full file path for each icon. Either:

```
/uk.ac.stfc.isis.ibex.ui.scripting/icons/script_32x24.png alarm icon 
designed by Freepik from www.flaticon.com

/uk.ac.stfc.isis.ibex.ui.scripting/icons/example_32x24.png alarm icon 
designed by Freepik from www.flaticon.com
```

Or, to save some space and be a bit less verbose

```
Beam Status
Located in uk.ac.stfc.isis.ibex.ui.beamstatus/icons/

status_24x24 designed by Freepik from www.flaticon.com
status_32x24 designed by Freepik from www.flaticon.com
```

Similarly to how you are doing it, just a little more obvious.

Comment (aaron-long) [reply base/uk.ac.stfc.isis.ibex.ui.help/resources/iconlicences.txt]: As per our conversation: I have updated the icon path to have a full directory path and placed these inline with the sub-headings.

File: base/uk.ac.stfc.isis.ibex.ui.help/src/uk/ac/stfc/isis/ibex/ui/help/IconLicencesDialogBox.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+     */
+	public IconLicencesDialogBox(Shell parentShell) {
+		super(parentShell);
+		setShellStyle(getShellStyle() | SWT.RESIZE | SWT.MAX);
+	}
+
+	@Override
+	protected void configureShell(Shell newShell) {
+		super.configureShell(newShell);
+		setTitle("IBEX Icon Licences");
```
Comment (Rory-Potter-Tessella) [inline base/uk.ac.stfc.isis.ibex.ui.help/src/uk/ac/stfc/isis/ibex/ui/help/IconLicencesDialogBox.java]: Could extract the title string out.

**Pull request merged**